src/components/EarnDashboard/components/PositionDetails/StakingPositionsByProvider.tsx

```diff
@@ -280,22 +280,34 @@ export const StakingPositionsByProvider: React.FC<StakingPositionsByProviderProp
       {
         Header: () => null,
         id: 'expander',
-        Cell: ({ row }: { row: RowProps }) => (
-          <Flex justifyContent='flex-end' width='full'>
-            <Button
-              variant='ghost'
-              size='sm'
-              colorScheme='blue'
-              width={widthMdAuto}
-              rightIcon={row.original.isReadOnly ? <ExternalLinkIcon boxSize={3} /> : undefined}
-              // we need to pass an arg here, so we need an anonymous function wrapper
-              // eslint-disable-next-line react-memo/require-usememo
-              onClick={() => handleClick(row, DefiAction.Overview)}
-            >
-              {translate(row.original.isReadOnly ? 'common.view' : 'common.manage')}
-            </Button>
-          </Flex>
-        ),
+        Cell: ({ row }: { row: RowProps }) => {
+          const url = getMetadataForProvider(row.original.provider)?.url
+          const translation = (() => {
+            if (!row.original.isReadOnly) return 'common.manage'
+            return url ? 'common.view' : undefined
+          })()
+
+          return (
+            <Flex justifyContent='flex-end' width='full'>
+              {translation && (
+                <Button
+                  variant='ghost'
+                  size='sm'
+                  colorScheme='blue'
+                  width={widthMdAuto}
+                  rightIcon={
+                    row.original.isReadOnly && url ? <ExternalLinkIcon boxSize={3} /> : undefined
+                  }
+                  // we need to pass an arg here, so we need an anonymous function wrapper
+                  // eslint-disable-next-line react-memo/require-usememo
+                  onClick={() => handleClick(row, DefiAction.Overview)}
```

Comment:
We can use a memoized closure here:

```suggestion
                  onClick={handleActionClick}
```

where `handleActionClick` is:

```ts
const handleActionClick = useCallback(() => handleClick(row, DefiAction.Overview), [row])
```

Reply:
Tackle all instances in 825fb6fe10 https://jam.dev/c/bad6d247-5a8e-417e-86bf-fc4f86e000a5, nice one @0xApotheosis 🍛 
Always keep on forgetting those cells are actual components and can use effects 🧠